odbc/src/handles.rs

```diff
@@ -0,0 +1,202 @@
+use std::{
+    collections::HashSet,
+    sync::{Mutex, RwLock},
+};
+
+use odbc_sys::Integer;
+
+#[derive(Debug)]
+pub struct EnvHandle {
+    pub env: RwLock<Env>,
+}
+
+impl EnvHandle {
+    pub fn new() -> Self {
+        Self {
+            env: RwLock::new(Env {
+                _attributes: EnvAttributes::default(),
+                state: EnvState::Unallocated,
+                connections: HashSet::new(),
+            }),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Env {
+    // attributes for this Env
+    pub _attributes: EnvAttributes,
```

Comment:
here and elsewhere in this patch, I'd suggest that we just omit things that aren't used yet, and instead add things in as we actually implement the things that use them

Reply:
We absolutely know we'll need attributes though. I'd be fine with removing the other things that aren't attributes/the handle state names.